### ISSUE TITLE
Ngalert: Alert/resolve notifications are sent unreliable

### ISSUE THREAD

**Issue by FireEmerald** [grafana/grafana#36644]
**What happened**:
The new ngalert system of grafana does not notify the configured _Contact points_ in a reliable manner. Testing it resulted in:
- Flipping alerts, delayed alerts, none alerts at all.

After alert creation and persistent constraint violation `(speed < 100)`, the POST endpoint received flipping alert notifications - although none state change happened in the db nor in the grafana interface (state was alerting after 30sec):

<details>

1. 06:45
``` json
{
"receiver": "autogen-panel-recv-default_route",
"status": "firing",
"startsAt": "2021-07-12T06:45:23.8635683Z",
"endsAt": "0001-01-01T00:00:00Z",
```
2. 06:46
``` json
"receiver": "autogen-panel-recv-default_route",
"status": "resolved",
"startsAt": "2021-07-12T06:45:23.8635683Z",
"endsAt": "2021-07-12T06:45:53.8635683Z",
```
3. 06:47
``` json
"receiver": "autogen-panel-recv-default_route",
"status": "firing",
"startsAt": "2021-07-12T06:45:23.8635683Z",
"endsAt": "0001-01-01T00:00:00Z",
```
4. 06:47
``` json
"receiver": "autogen-panel-recv-default_route",
"status": "resolved",
"startsAt": "2021-07-12T06:45:23.8635683Z",
"endsAt": "2021-07-12T06:47:03.8635683Z",
```
(...)
</details>

After changing the speed back to `>100` in the db, the flipping error stopped. Then i changed the `Group interval: 5m` and didn't received any notification anymore. Also changing the value back to `2s` changed nothing.

**What you expected to happen**:
When a alert is triggered there should be a immediate notification send to the configured _Contact point_. If none _simulated_ delay is configured. Also the _resolved_ notification should be send within seconds if configured.

**How to reproduce it (as minimally and precisely as possible)**:
<details>

1. Set the default notification endpoint to webhook and POST requests
2. Attach a postgres database to grafana containing the following table:
``` sql
CREATE TABLE "carmap" (
  "id" bigserial NOT NULL,
  "captured" timestamp(6) NOT NULL,
  "name" varchar(255) NOT NULL,
  "city" varchar(255) NOT NULL,
  "speed" float8 NOT NULL
);

-- Insert some rows, adjust speed (->50) to trigger alerts
INSERT INTO carmap (captured, name, city, speed) VALUES
  (NOW()-INTERVAL '3 SECOND', 'car1', 'berlin', 150),
  (NOW()-INTERVAL '3 SECOND', 'car2', 'berlin', 130);
```
3. Create a new alert, sql:
``` sql
SELECT t.name, t.city, CASE WHEN t.speed > 100 THEN 0 ELSE 1 END
FROM (
  SELECT RANK()
    OVER(
      PARTITION BY name
      ORDER BY captured DESC
    ) AS rnk,
    id, captured, name, city, speed
  FROM carmap
) t
WHERE
  rnk = 1 AND
  captured >= (NOW() - INTERVAL '5 MIN')
```
![2021-07-12 09_11_58-Alerting_ Alert rules - Grafana](https://user-images.githubusercontent.com/1107939/125245913-6e45dc80-e2f1-11eb-98ee-ad3d5d9db306.png)

4. Alert conditions:
![2021-07-12 09_12_20-Alerting_ Alert rules - Grafana](https://user-images.githubusercontent.com/1107939/125245887-68e89200-e2f1-11eb-82a2-617e1be21f8c.png)
</details>

**Anything else we need to know?**:
- Contact point: autogen-panel-recv-default_route
- Group by: -
- Timings: Group wait: 1s | Group interval: 2s | Repeat interval: 15m

**Environment**:
- Grafana version: v8.0.3 (cae5c5e46b)
- Data source type & version: postgres 13.3
- OS Grafana is installed on: linux
- User OS & Browser: windows, chrome

**Comment by sparkstar** [comment]
I also have the same problem. ngalert is flipping 30/40 seconds periodically. delayed too.

```json
16/Jul/2021 07:49:22 {"receiver": "...", "status": "resolved", "alerts": [{"endsAt": "2021-07-16T07:48:24.503642489Z",...}]}
16/Jul/2021 07:49:52 {"receiver": "...", "status": "firing", "alerts": [{"endsAt": "0001-01-01T00:00:00Z",...}]}
16/Jul/2021 07:50:22 {"receiver": "...", "status": "resolved", "alerts": [{"endsAt": "2021-07-16T07:49:44.503642489Z",...}]}
16/Jul/2021 07:51:01 {"receiver": "...", "status": "firing", "alerts": [{"endsAt": "0001-01-01T00:00:00Z",...}]}
...
```

I got a notify via Webhook, but Mail or Telegram is same. and Alert Dashboard show valid alerting state when during flipped.
Grafana Version I used 8.0.3 ~ 8.0.6.

**Comment by codesome** [comment]
> When a alert is triggered there should be a immediate notification send to the configured Contact point. If none simulated delay is configured.

Right! We had some fixed wait for all notifications, which should not actually be present. https://github.com/grafana/grafana/pull/37203 removes that fixed wait and this fix should be available in 8.1.0.

**Comment by codesome** [comment]
@FireEmerald @sparkstar would it be possible to test the `main` branch and see if the issue if fixed now? (other option is to wait for 8.1.0 which is due too)

**Comment by sparkstar** [comment]
> @FireEmerald @sparkstar would it be possible to test the `main` branch and see if the issue if fixed now? (other option is to wait for 8.1.0 which is due too, the 8.1.0-beta2 does not have the possible fix)

I checked your commit and fix config directly. In grafana.ini, Alerting `notification_timeout_seconds` set to 0. and I didn't receive delayed alert anymore.

I use grafana 8.1.0-beta2 now, and test `main` branch tomorrow. Thank you!

**Comment by sparkstar** [comment]
I tested grafana tag `main-ubuntu` today, didn't receive delayed alert too.

**Comment by armandgrillet** [comment]
This issue should be fixed in Grafana 8.1 (which has been released on August 5th), please reopen the issue if this is not the case.
